### ISSUE TITLE
Add pickle support

### ISSUE THREAD

**Issue by snorfalorpagus** [jetuk/pycatchmod#4]
It should be easy to add pickle support to the classes. This could be tied into read/write support for JSON.

(I realise `catchment_from_json()` exists - I was thinking, `Catchment.load()` and `Catchment.dump()`.

**Comment by jetuk** [comment]
Would `load` and `dump` use a JSON format?

**Comment by snorfalorpagus** [comment]
Yes. I was thinking that `load` and `dump` handle the dictionary of parameters. The `__setstate__` and `__getstate__` methods could then make use of this, plus a little extra to handle the actual state.
